### ISSUE TITLE
Query / discussion about the experimental setups.

### ISSUE THREAD

**Issue by zhaisf** [Stry233/SAMA#1]
Dear Authors:

Thanks for the great and inspiring work. I have been following this for a while. Congrats on the ICLR acceptance.

However, I ran into some issues while trying to reproduce the experiments. 

1. In the paper you mention _"random mask configurations with 15% of tokens"_, but in loss.py (https://github.com/Stry233/SAMA/blob/d19e4d4a70a31870ccbd5031d6e9bc6235f1028a/attack/attacks/loss.py#L11C8-L11C67) the implementation is 
`dataset = dataset.map(lambda x: {self.name: -x["nlloss"]})
`
This seems different from the paper. It looks like the mask ratio becomes 100%, which is not the same as the loss attack and may make it harder to capture membership signals.

    Also, when we strictly follow the loss-attack setup (with a reference model) and sample on the ELBO loss with mask ratio 0.3~0.5, we get strong membership inference results, sometimes even better than SAMA. Have you tried this setting?

2. In the paper, your main experiments use the MIMIR pre-training MIA dataset. Note that this dataset targets MIA evaluation at the pretraining stage, so an **unfinetuned model may already have some prior knowledge** of it. For example, when we run a loss MIA on an unfinetuned LLaDA-base model, we get AUC scores of 0.80 on the GitHub dataset and 0.68 on the arXiv dataset (ngram_7_0.2).  

    This may raise the concern that MIMIR may not be suitable as an evaluation dataset for MIA at the fine-tuning stage.

I would like to discuss these points with you to clear up any potential misunderstandings on my side. 

Looking forward to your response.

Best,
Shengfang

**Comment by Stry233** [comment]
Hi Shengfang.

Thanks so much for the kind words and for taking the time to dig into the code and share your findings! Really appreciate it. Let me address your points one by one. I don't regularily check the issue page of this repo so sorry for the late reply. For future discussion, feel free to cc them to my email `yuetian@purdue.edu` if you can.

**1. On the `dataset.map(lambda x: {self.name: -x["nlloss"]})` line**

Totally understand the confusion here, and I admit the code could use better commenting. What's happening is that most of the baselines (borrowed from the black-box LLM MIA literature, see the first line of each attack module for their original references) operate on sequence-level loss. Since they all draw from the same signal source, it would be painfully slow to recompute the loss every time for every attack. So instead, I cache the loss into a signal pool, and each attack just pulls from `x["nlloss"]` as needed.

This is definitely not a 100% mask ratio. The actual loss computation happens earlier in the pipeline: specifically at line 90 of `attack/run.py` and line 43 of `attack/misc/dataset.py`, which call the loss function in `attack/attacks/utils.py`. That part follows LLaDA's own implementation, so you can cross-reference their repo for the details. But yes, you're right that this deserves clearer documentation, and I appreciate you flagging it.

**2. On getting strong results with loss + reference model at mask ratio 0.3–0.5**

This is interesting and I'd love to hear more about the specifics. We actually tested a wide range of mask ratios internally during early development but never observed consistently better results in the way you describe. A few things I'd be curious about:

- What's your training recipe? In my experience, results like the ones you mention often come from an underfit model where the domain adaptation signal hasn't yet contaminated the membership signal (we discuss this dynamic in Appendix E). This matters a lot in the fine-tuned model setting, and it's actually one of the key motivations behind why we frame MIA as a robust statistics problem and use local subset sampling.
- How large is the gap you're seeing, and does the trend hold consistently across datasets and seeds in high fidelity metric as well?

Would be great to compare notes on this.

**3. On MIMIR as an evaluation dataset for fine-tuning stage MIA**

You're absolutely right, and I fully agree. MIMIR was designed for the pretraining MIA paradigm, and it's generally too small to be practical for evaluating fine-tuning stage attacks. We included it mainly because some prior works used it in the fine-tuned setting, and we wanted to keep comparisons fair. That's also why we included other, more practical datasets like WikiText-103 (see Appendix D.3).

As for why splits like GitHub show high AUC even on an unfinetuned model: my intuition is that this is largely a dataset artifact. Code is highly structured, so n-gram deduplication ends up creating a bigger distributional gap between members and non-members. You can actually pick up on this without any model access at all. If you run the Bag-of-Words baseline, you'll see AUC > 0.5, which we show in the paper. That's a red flag, because BoW doesn't touch the model.

On a broader note, I think the future of MIA evaluation in the fine-tuning setting needs to move toward more practical datasets. For instance, in my next work ([arXiv:2601.02751](https://arxiv.org/pdf/2601.02751), appearing at USENIX '26), we use datasets like Cosmopedia, which is large (so it's practical), AI-generated, and new (so it's unlikely to be in pretraining data). We also ran some small-scale tests on similar datasets for SAMA after we archive this work, and the results looked solid, but there's definitely more to be done here.

Thanks again for the thoughtful questions, Shengfang. Hope this clears things up! I will close this issue just for a clearer tracking, but feel free to reopen for any discussion later.

Best,
Yuetian
